backends/p4tools/testgen/core/small_step/expr_stepper.cpp

```diff
@@ -0,0 +1,475 @@
+    : AbstractStepper(state, solver, programInfo) {}
+
+bool ExprStepper::preorder(const IR::BoolLiteral* boolLiteral) {
+    logStep(boolLiteral);
+    return stepSymbolicValue(boolLiteral);
+}
+
+bool ExprStepper::preorder(const IR::Constant* constant) {
+    logStep(constant);
+    return stepSymbolicValue(constant);
+}
+
+void ExprStepper::handleHitMissActionRun(const IR::Member* member) {
+    auto* nextState = new ExecutionState(state);
+    std::vector<Continuation::Command> replacements;
+    const auto* method = member->expr->to<IR::MethodCallExpression>();
+    BUG_CHECK(method->method->is<IR::Member>(), "Method apply has unexpected format: %1%", method);
+    replacements.emplace_back(new IR::MethodCallStatement(method));
+    const auto* methodName = method->method->to<IR::Member>();
+    const auto* table = state.getTableType(methodName);
+    CHECK_NULL(table);
+    if (member->member.name == IR::Type_Table::hit) {
+        replacements.emplace_back(Continuation::Return(TableStepper::getTableHitVar(table)));
+    } else if (member->member.name == IR::Type_Table::miss) {
+        replacements.emplace_back(
+            Continuation::Return(new IR::LNot(TableStepper::getTableHitVar(table))));
+    } else {
+        replacements.emplace_back(Continuation::Return(TableStepper::getTableActionVar(table)));
+    }
+    nextState->replaceTopBody(&replacements);
+    result->emplace_back(nextState);
+}
+
+bool ExprStepper::preorder(const IR::Member* member) {
+    logStep(member);
+
+    if (member->expr->is<IR::MethodCallExpression>() &&
+        (member->member.name == IR::Type_Table::hit ||
+         member->member.name == IR::Type_Table::miss ||
+         member->member.name == IR::Type_Table::action_run)) {
+        handleHitMissActionRun(member);
+        return false;
+    }
+
+    // TODO: Do we need to handle non-numeric, non-boolean expressions?
+    BUG_CHECK(member->type->is<IR::Type::Bits>() || member->type->is<IR::Type::Boolean>() ||
+                  member->type->is<IR::Extracted_Varbits>(),
+              "Non-numeric, non-boolean member expression: %1% Type: %2%", member,
+              member->type->node_type_name());
+
+    // Check our assumption that this is a chain of member expressions terminating in a
+    // PathExpression.
+    checkMemberInvariant(member);
+
+    // This case may happen when we directly assign taint, that is not bound in the symbolic
+    // environment to an expression.
+    if (SymbolicEnv::isSymbolicValue(member)) {
+        return stepSymbolicValue(member);
+    }
+    return stepSymbolicValue(state.get(member));
+}
+
+void ExprStepper::evalActionCall(const IR::P4Action* action, const IR::MethodCallExpression* call) {
+    const auto* actionNameSpace = action->to<IR::INamespace>();
+    BUG_CHECK(actionNameSpace, "Does not instantiate an INamespace: %1%", actionNameSpace);
+    auto* nextState = new ExecutionState(state);
+    if (action->body == nullptr) {
```

Comment:
I don't think this can happen

Reply:
I believe we ran into this. If not I will remove it but I need to check.